core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.spec.ts

```diff
@@ -0,0 +1,883 @@
+    beforeEach(async(() => {
+      mockResizeEmitter = new EventEmitter();
+      TestBed.configureTestingModule({
+        imports: [
+          BrowserAnimationsModule,
+          MaterialModule,
+          FormsModule,
+          HttpClientTestingModule
+        ],
+        declarations: [
+          NewLearnerDashboardComponent,
+          MockTranslatePipe,
+          SortByPipe,
+          MockSlicePipe,
+          MockTrunctePipe,
+          BackgroundBannerComponentStub,
+          ExplorationSummaryTileComponentStub,
+          CollectionSummaryTileComponentStub,
+          LoadingDotsComponentStub,
+        ],
+        providers: [
+          AlertsService,
+          DateTimeFormatService,
+          ExplorationObjectFactory,
+          LearnerDashboardBackendApiService,
+          {
+            provide: LearnerDashboardActivityBackendApiService,
+            useClass: MockLearnerDashboardActivityBackendApiService
+          },
+          {
+            provide: WindowDimensionsService,
+            useValue: {
+              isWindowNarrow: () => true,
+              isTabletView: () => true,
+              getResizeEvent: () => mockResizeEmitter,
+            }
+          },
+          SuggestionModalForLearnerDashboardService,
+          UrlInterpolationService,
+          UserService,
+          PageTitleService,
+          {
+            provide: TranslateService,
+            useClass: MockTranslateService
+          }
+        ],
+        schemas: [NO_ERRORS_SCHEMA]
+      }).compileComponents();
+    }));
+
+    beforeEach(fakeAsync(() => {
+      fixture = TestBed.createComponent(NewLearnerDashboardComponent);
+      component = fixture.componentInstance;
+
+      alertsService = TestBed.inject(AlertsService);
+      csrfTokenService = TestBed.inject(CsrfTokenService);
+      dateTimeFormatService = TestBed.inject(DateTimeFormatService);
+      explorationObjectFactory = TestBed.inject(ExplorationObjectFactory);
+      windowDimensionsService = TestBed.inject(WindowDimensionsService);
+      learnerDashboardBackendApiService =
+        TestBed.inject(LearnerDashboardBackendApiService);
+      suggestionModalForLearnerDashboardService =
+        TestBed.inject(SuggestionModalForLearnerDashboardService);
+      userService = TestBed.inject(UserService);
+      translateService = TestBed.inject(TranslateService);
+      pageTitleService = TestBed.inject(PageTitleService);
+      learnerGroupBackendApiService = TestBed.inject(
+        LearnerGroupBackendApiService);
+
+      const mockElement = document.createElement('div');
+      mockElement.className = 'oppia-exploration-title';
+      document.body.appendChild(mockElement);
+
+      spyOn(csrfTokenService, 'getTokenAsync').and.callFake(async() => {
+        return Promise.resolve('sample-csrf-token');
+      });
+      // Generate completed explorations and exploration playlist.
+      for (let i = 0; i < 10; i++) {
```

Comment:
What is the significance of `10` here? Would it be better to add it as a variable?

Reply:
Hi, @Lawful2002 Thanks for reviewing. I have created **new-learner-dashboard.ts/spec.ts**  copy of **learner-dashboard.ts/spec.ts** to maintain the existing learner-dashboard and redesigned learner-dashboard. So I have not written this test. It is already written Please see here (https://github.com/SubhashThenua/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts#L350)
Thanks!